### PR TITLE
[2/k][ui] Replace compute kind filters with kind filters

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Re-implementation of #23805 which is slightly lighter weight (but still touches a lot of surface area). Substitutes in the notion of a generic kind filter rather than specifically compute kind.

## Test Plan

Existing unit tests, local dagit.

## Changelog 

`NOCHANGELOG`

### CODE REVIEW

**Comment by benpankow** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24160?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#24160** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24160?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#24159** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24159?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23779** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/23779?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#24154](https://github.com/dagster-io/dagster/pull/24154) <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24154?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @benpankow and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetTagFilter.tsx [hunk 1]
```diff
@@ -98,3 +75,26 @@ export const BaseConfig: StaticBaseConfig<DefinitionTag> = {
   getStringValue: ({value, key}: DefinitionTag) => `${key}: ${value}`,
   matchType: 'all-of',
 };
+
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetTagFilter.tsx:78]: undo?

File: js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetTagFilter.tsx [hunk 2]
```diff
@@ -98,3 +75,26 @@ export const BaseConfig: StaticBaseConfig<DefinitionTag> = {
...
+        .map((jsonTag) => buildDefinitionTag(JSON.parse(jsonTag)))
+        .sort((a, b) =>
+          // Sort by key then by value
+          a.key.localeCompare(b.key) === 0
+            ? a.value.localeCompare(b.value)
+            : a.key.localeCompare(b.key),
+        ),
+    [assets],
+  );
+}
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/ui/Filters/useAssetTagFilter.tsx:100]: I don't see any changes here?

File: js_modules/dagster-ui/packages/ui-core/src/assets/useAssetCatalogFiltering.oss.tsx [hunk 3]
```diff
@@ -152,7 +152,6 @@ export function useAssetCatalogFiltering<
     [
       ['owners', filters.owners, allAssetOwners] as const,
       ['tags', filters.tags, tags] as const,
-      ['kinds', filters.kinds, allKinds] as const,
```
Comment (salazarm) [inline js_modules/dagster-ui/packages/ui-core/src/assets/useAssetCatalogFiltering.oss.tsx]: why did we remove this?

**Review by salazarm — APPROVED** [review]

**Pull request merged**